### PR TITLE
stbt.ocr: Always enable cache so we don't call tesseract again for the same pixels & parameters

### PR DESCRIPTION
**Pull request by drothlis**
We already have a way of caching image-processing operations (such as `match` and `ocr`) but we can either cache them all or not at all. This allows you to selectively enable the cache at run-time for *some* ocr operations.

TODO:

- [x] Decide on the API.
- [x] Tests.

### CODE REVIEW

**Comment by drothlis** [comment]
@wmanley please review the API.

- In the first commit I added the `stbt.enable_caching()` contextmanager.

- In the second commit I simplified the implementation by breaking backwards compatibility of the `imgproc_cache` API. This was always a private API so I think it's worth it.

- In the last commit I added a `cache=<bool>` parameter to `stbt.ocr()`. I'm not so sure about this one, might drop it. I expect the caching to be useful in a very limited set of cases, so it doesn't matter if it's a bit more verbose. In those cases you would do this:

  ```
  with stbt.enable_caching():
      return stbt.ocr(self._frame, region=...)
  ```

**Comment by drothlis** [comment]
From discussion with @wmanley:

* Enable ocr caching by default. This would allow us to use OCR more freely for menu navigation, in `wait_until` loops, etc.

**Comment by drothlis** [comment]
Open questions:

- [ ] Should we remove stbt.ocr's `cache` parameter? (And leave it so that it's always caching.) I'd rather not have a parameter that's there "just in case", is never used, but we have to continue to support & document it. Another disadvantage is that `cache=False` will apply to the Object Repository too, where we *know* it's safe to use.
- [ ] Should we add caching to `match_text` too?
- [ ] Should we remove the `with stbt.enable_caching()` public API I've added? I don't know any compelling use-cases apart from ocr.
- [ ] Regarding the implementation: Maybe instead of the extra `enable_caching` contextmanager, the `memoize` decorator could inspect the arguments to see if one of them is called `cache`, and if False, disable the caching?

**Comment by drothlis** [comment]
Rebased onto master and squashed away intermediate commits where I changed my mind multiple times -- to make reviewing easier.

**Review by wmanley — CHANGES_REQUESTED** [review]

File: _stbt/imgproc_cache.py [hunk 1]
```diff
@@ -69,6 +80,19 @@ def cache(filename=None):
...
+    global _enabled
+    previous_value = _enabled
+    try:
+        _enabled = enable
+        yield
+    finally:
+        _enabled = previous_value
+
+
 def memoize(additional_fields=None):
```
[the remark this answers is not in the document]
Comment (drothlis) [reply _stbt/imgproc_cache.py:104]: Huh, the thought of thread safety never even crossed my mind. I have added `use_cache` to the memoized function (`inner`) in 6574bcfc.

**Pull request merged**